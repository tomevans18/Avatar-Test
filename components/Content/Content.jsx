import React from 'react';
import PropTypes from 'prop-types';

import Div from './Content.style';

const propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
};

const Content = props => {
  const { children, ...otherProps } = props;
  return <Div {...otherProps}>{children}</Div>;
};

Content.propTypes = propTypes;

export default Content;
