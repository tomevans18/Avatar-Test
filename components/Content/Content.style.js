import styled from 'styled-components';
import media from '../../styles/lib/mediaQueries';

export default styled.div`
  flex: 1;
  flex-direction: column;
  outline: none;

  ${media.medium`
    display: flex;
  `};
`;
