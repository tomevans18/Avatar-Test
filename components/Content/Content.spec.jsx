import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Content from './Content';

const passedProps = {
  passedComponentTest: 'test string',
};

describe('Content', () => {
  it('should render as expected ', () => {
    const wrapper = mount(
      <Content {...passedProps}>
        <div>TEST CONTENT</div>
      </Content>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
