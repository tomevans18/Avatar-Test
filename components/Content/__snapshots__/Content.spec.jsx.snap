// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render as expected  1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  outline: none;
}

@media (min-width:769px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<Content>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  flex: 1;
  flex-direction: column;
  outline: none;

  ",
              "
    @media (min-width: ",
              "769",
              "px) {
      ",
              "
    display: flex;
  ",
              "
    }
    ",
              "
  ",
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <div>
          TEST CONTENT
        </div>
      </div>
    </StyledComponent>
  </styled.div>
</Content>
`;
